### PR TITLE
feat: avatar selection and store as profile

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

This PR adds:

- avatar selection from the user NFT list - stored in the newly added nft slice, and keyed by walletId so we don't store across wallets
- the first bit of the `nftApi`, for now only with `selectedNftAvatarByWalletId` i.e

```typescript
type NftState = {
  selectedNftAvatarByWalletId: Record<WalletId, AssetId>
}
```
- whitelists the newly added API so we persist across refreshes

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

We might be overfetching NFTs and rugging our rate limiting for Zapper/Alchemy/Covalent

## Testing

- The NFT asset selection modal displays your currently available NFTs
- It is possible to select and set an NFT as avatar
- Refreshing the page keeps the NFT selected
- Switches from one wallet to another remembers the selected NFT for said wallet
- It is possible to select the default blockies image again in case you change your mind and don't want to have an NFT as an avatar anymore

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4563** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4563" target="_blank">Graphite</a> 
    * **PR #4574** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4574" target="_blank">Graphite</a> 
      * **PR #4582** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4582" target="_blank">Graphite</a> 
        * **PR #4575** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4575" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4575?utm_source=stack-comment).

File: src/components/Modals/Nfts/NftModal.tsx [hunk 1]
```diff
@@ -117,6 +119,17 @@ export const NftModal: React.FC<NftModalProps> = ({ nftItem }) => {
       } as const),
     [],
   )
+
+  const handleSetAsAvatarClick = useCallback(() => {
+    // Unable to get the AssetId of the collection, this should never happen but it may
+    // TODO(gomes): remove nftAssetId manual serialization when we have a normalized nft slice with nft id as AssetId
+    //
```
Comment (gomesalexandre) [inline src/components/Modals/Nfts/NftModal.tsx:126]:
```suggestion
```

File: src/state/apis/nft/nftApi.ts [hunk 2]
```diff
@@ -28,6 +31,33 @@ type GetNftCollectionInput = {
 
 const moduleLogger = logger.child({ namespace: ['nftApi'] })
 
+type NftState = {
+  selectedNftAvatarByWalletId: Record<WalletId, string>
```
Comment (gomesalexandre) [inline src/state/apis/nft/nftApi.ts:35]:
```suggestion
  selectedNftAvatarByWalletId: Record<WalletId, AssetId>
```

Comment (gomesalexandre) [reply src/state/apis/nft/nftApi.ts]: 4b6c0c3524dbbb64acdbd5401364754dbd1b03f3

**Review by 0xApotheosis — APPROVED** [review]
What a great PR 🤝 

Tested NFT selection across multiple wallets, and everything worked as expected.

Screenshot 2023-06-01 at 3 50 53 pm

Screenshot 2023-06-01 at 3 51 17 pm

Screenshot 2023-06-01 at 3 51 31 pm

**Pull request merged**